### PR TITLE
ena's blue-hat solution

### PR DESCRIPTION
**Pull request by enarha**

### CODE REVIEW

File: blue-hat/ena/hat.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+func main() {
+
+	const sourceFile = "red_hat.png"
+	const outputFile = "modified_hat.png"
+
+	fin, err := os.Open(sourceFile)
+	defer func() { fin.Close() }()
+	if err != nil {
+		panic(err)
```
Comment (davidmogar) [inline blue-hat/ena/hat.go:22]: This prints the stack trace. Not sure this is what you want.

Comment (enarha) [reply blue-hat/ena/hat.go:22]: I know what it does.

File: blue-hat/ena/hat.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+	// registration of the png format
+	_ "image/png"
+	"os"
+)
+
+func main() {
+
+	const sourceFile = "red_hat.png"
+	const outputFile = "modified_hat.png"
```
[the remark this answers is not in the document]
Comment (enarha) [reply blue-hat/ena/hat.go:17]: Why?

File: blue-hat/ena/hat.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+			rgba := color.RGBAModel.Convert(img.At(x, y)).(color.RGBA)
+			newImg.Set(x, y, color.RGBA{R: rgba.B, G: rgba.G, B: rgba.R, A: rgba.A})
+		}
+	}
+
+	fout, err := os.Create(outputFile)
+	if err != nil {
+		panic(err)
+	}
+	defer func() { fout.Close() }()
```
[the remark this answers is not in the document]
Comment (enarha) [reply blue-hat/ena/hat.go:43]: I know. I was experimenting with os.exit() as part of the defer.

File: blue-hat/ena/hat.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (davidmogar) [reply blue-hat/ena/hat.go:17]: This comment comes probably from what I usually do and I'm accustomed to see in Python. Absolutely nothing wrong with your code and maybe is the best right way to do it.

**Pull request merged**